modules/ROOT/pages/traversal-framework/traversal_framework_java_api.adoc

```diff
@@ -0,0 +1,353 @@
+* `BranchOrderingPolicies.PREORDER_BREADTH_FIRST` -- Traversing breadth first, visiting each node before visiting its child nodes.
+* `BranchOrderingPolicies.POSTORDER_BREADTH_FIRST` -- Traversing breadth first, visiting each node after visiting its child nodes.
+
+[NOTE]
+====
+Breadth-first traversals have a higher memory overhead than depth-first traversals.
+====
+
+A `BranchSelector` carries state and hence needs to be uniquely instantiated for each traversal.
+Therefore it is supplied to the `TraversalDescription` through a `BranchOrderingPolicy` interface, which is a factory of `BranchSelector` instances.
+
+A user of the Traversal Framework rarely needs to implement their own `BranchSelector` or `BranchOrderingPolicy`, it is provided to let graph algorithm implementors provide their own traversal orders.
+The Neo4j Graph Algorithms package contains for example a `BestFirst` order `BranchSelector` / `BranchOrderingPolicy` that is used in BestFirst search algorithms such as A* and Dijkstra.
+
+=== Example
+Given the following graph, these are the results of the branch ordering policies without any extra filter:
+
+image::traversal_order_example_graph.png[align="center", role="middle", width=200]
+
+[cols="1,1"]
+|===
+|Ordering policy |Order of the nodes in traversal
+
+|`BranchOrderingPolicies.PREORDER_DEPTH_FIRST`
+|a, b, d, c, e
+
+|`BranchOrderingPolicies.POSTORDER_DEPTH_FIRST`
+|d, b, e, c, a
+
+|`BranchOrderingPolicies.PREORDER_BREADTH_FIRST`
+|a, b, c, d, e
+
+|`BranchOrderingPolicies.POSTORDER_BREADTH_FIRST`
+|d, e, b, c, a
+|===
+
+=== BranchOrderingPolicy
+
+A `BranchOrderingPolicy` is a factory for creating ``BranchSelector``s to decide in what order branches are returned
+(where a branch's position is represented as a `Path` from the start node to the current node).
+
+Depth-first and breadth-first are common policies and can be accessed by the convenience methods breadthFirst() / depthFirst().
+
+==== Example
+[source, java, role="nocopy"]
+----
+TraversalDescription td;
+try ( Transaction tx = graphDb.beginTx() ) {
+     td = tx.traversalDescription();
+            .depthFirst()
+}
+
+td.traverse( startNode );
+----
+
+This is equivalent to setting the `BranchOrderingPolicies.PREORDER_DEPTH_FIRST` / `BranchOrderingPolicies.PREORDER_BREADTH_FIRST` policy.
+
+[source, java, role="nocopy"]
+----
+TraversalDescription td;
+try ( Transaction tx = graphDb.beginTx() ) {
+     td = tx.traversalDescription();
+            .order( BranchOrderingPolicies.PREORDER_BREADTH_FIRST )
+}
+
+td.traverse( startNode );
+----
+
+[[traversal-java-api-pathexpander]]
+== PathExpander
+
+The Traversal Framework uses a `PathExpander` to discover the relationships
+that should be followed from a particular path to further branches in the traversal.
+
+There are multiple ways of specifying a `PathExpander`:
+
+* The built-in `PathExpander` define some commonly used ``PathExpander``s.
+* The `PathExpanderBuilder` allows the combination of definitions.
```

Comment:
I don't know how useful is to have the built-in ones in this case, cause the syntax of `PathExpanderBuilder` is quite succinct already and it lets you do more combinations.

Reply:
We think it is best to document all that is there to give the user a full overview, so omitting the "convenience" methods seems strange.